src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +105,55 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                ret._forwardingApp = new MSBuildForwardingApp(msbuildArgs, msbuildPath);
+
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (ret._forwardingApp == null)
+            {
+                throw new CodeNotExecutedException(exitCode);
+            }
+
+            return ret;
+        }
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            BuildCommand cmd;
+            try
+            {
+                cmd = FromArgs(args);
+            }
+            catch (CodeNotExecutedException e)
```

Comment:
CodeNotExecutedException is a little strange to me because we throw it before we attempt to execute (we attempt to execute later in this method). To me we don't need the exception, just check for null.